src/server/game/Entities/Player/Player.cpp

```diff
@@ -1776,6 +1777,14 @@ void Player::Update(uint32 p_time)
     {
         if (p_time >= m_zoneUpdateTimer)
         {
+            // On zone update tick check if we are still in an inn if we are supposed to be in one
```

Comment:
really no better way can be found??

Reply:
ok rochet2 the boss enlightened me on why the client can't be trusted when exiting the inn, hence why there is prob no other efficient way to check when out of the inn. Client could never send the "not in inn anymore" and so players would cheat the resting system forever

so i guess no better way

Reply:
This is needed, else you would be back at square one of the rest flags not updating again.